From 21c0c4e29521ee94bbb44f16c7f55c1b76088135 Mon Sep 17 00:00:00 2001
Message-ID: <21c0c4e29521ee94bbb44f16c7f55c1b76088135.1726926634.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1726926633.git.sharlatanus@gmail.com>
References: <cover.1726926633.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Fri, 20 Sep 2024 22:42:56 +0100
Subject: [PATCH v2 24/25] gnu: Add python-pytest-cython.

* gnu/packages/python-check.scm (python-pytest-cython): New variable.

Change-Id: I4f223bb0a88db3fd91bb312dfe4558a075ff8bb2
---
 gnu/packages/python-check.scm | 38 +++++++++++++++++++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/gnu/packages/python-check.scm b/gnu/packages/python-check.scm
index 4888b42bc1..382a5453bb 100644
--- a/gnu/packages/python-check.scm
+++ b/gnu/packages/python-check.scm
@@ -1784,6 +1784,44 @@ (define-public python-pytest-celery
      "This package provides a shim Pytest plugin to enable a Celery marker.")
     (license license:bsd-3)))
 
+(define-public python-pytest-cython
+  (package
+    (name "python-pytest-cython")
+    (version "0.3.1")
+    (source
+     (origin
+       (method url-fetch)
+       (uri (pypi-uri "pytest-cython" version))
+       (sha256
+        (base32 "0ma496dgmmrpgqd3zk6vin29dgajcplh63yqd8jh2a3ai954fr22"))))
+    (build-system pyproject-build-system)
+    (arguments
+     (list
+      #:test-flags
+      #~(list "tests"
+              ;; FIXME: Failed: nomatch: '*sqr*PASSED*
+              "-k" (string-append
+                    "not test_wrap_cpp_ext_module[importlib]"
+                    " and not test_wrap_c_ext_module[importlib]"
+                    " and not test_cython_ext_module[importlib]"))
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-before 'check 'build-extensions
+            (lambda _
+              (with-directory-excursion "tests/example-project"
+                (invoke "python" "setup.py" "build_ext" "--inplace")))))))
+    (native-inputs
+     (list python-nox
+           python-cython-3
+           python-setuptools))
+    (propagated-inputs
+     (list python-pytest-8))
+    (home-page "https://github.com/lgpage/pytest-cython")
+    (synopsis "Cython extension modules testing plugin")
+    (description
+     "This package provides a plugin for testing Cython extension modules.")
+    (license license:expat)))
+
 (define-public python-pytest-env
   (package
     (name "python-pytest-env")
-- 
2.46.0

